services/core/shared/core/compat/sdk_v5/accounts.js

```diff
@@ -432,7 +434,51 @@ const getMultisignatureGroups = async account => {
 	return multisignatureAccount;
 };
 
-const getMultisignatureMemberships = async () => []; // TODO
+const getMultisignatureMemberships = async account => {
+	const multisignatureMemberships = {};
+	const multisignatureDB = await getMultisignatureIndex();
+	const memberInfo = await multisignatureDB.find({ memberAddress: account.address });
+
+	if (Array.isArray(memberInfo) && memberInfo.length) {
+		multisignatureMemberships.memberships = [];
+		await BluebirdPromise.map(
+			memberInfo,
+			async member => {
+				const result = await getIndexedAccountInfo({ address: member.groupAddress });
+				multisignatureMemberships.memberships.push({
+					address: result && result.address ? result.address : undefined,
+					username: result && result.username ? result.username : undefined,
+					publicKey: result && result.publicKey ? result.publicKey : undefined,
+				});
+			},
+			{ concurrency: memberInfo.length },
+		);
+	}
+	return multisignatureMemberships;
+};
+
+const resolveMultisignatureMemberships = tx => {
+	const multisignatureInfoToIndex = [];
+	tx.asset.mandatoryKeys.forEach(key => {
+		const members = {
+			id: getBase32AddressFromPublicKey(tx.senderPublicKey)
+				.concat('_', getBase32AddressFromPublicKey(key)),
+			memberAddress: getBase32AddressFromPublicKey(key),
+			groupAddress: getBase32AddressFromPublicKey(tx.senderPublicKey),
+		};
+		multisignatureInfoToIndex.push(members);
+	});
+	tx.asset.optionalKeys.forEach(key => {
+		const members = {
+			id: getBase32AddressFromPublicKey(tx.senderPublicKey)
+				.concat('_', getBase32AddressFromPublicKey(key)),
+			memberAddress: getBase32AddressFromPublicKey(key),
+			groupAddress: getBase32AddressFromPublicKey(tx.senderPublicKey),
+		};
+		multisignatureInfoToIndex.push(members);
+	});
+	return multisignatureInfoToIndex;
+};
```

Comment:
Please consider refactoring the code as below to avoid duplication of the logic.

```suggestion
const resolveMultisignatureMemberships = tx => {
	const multisignatureInfoToIndex = [];
	const allKeys = tx.asset.mandatoryKeys.concat(tx.asset.optionalKeys);
	allKeys.forEach(key => {
		const members = {
			id: getBase32AddressFromPublicKey(tx.senderPublicKey)
				.concat('_', getBase32AddressFromPublicKey(key)),
			memberAddress: getBase32AddressFromPublicKey(key),
			groupAddress: getBase32AddressFromPublicKey(tx.senderPublicKey),
		};
		multisignatureInfoToIndex.push(members);
	});
	return multisignatureInfoToIndex;
};
```
or 

```suggestion
const resolveMultisignatureMemberships = tx => {
	const multisignatureInfoToIndex = [];
	tx.asset.mandatoryKeys
		.concat(tx.asset.optionalKeys)
		.forEach(key => {
			const members = {
				id: getBase32AddressFromPublicKey(tx.senderPublicKey)
					.concat('_', getBase32AddressFromPublicKey(key)),
				memberAddress: getBase32AddressFromPublicKey(key),
				groupAddress: getBase32AddressFromPublicKey(tx.senderPublicKey),
			};
			multisignatureInfoToIndex.push(members);
		});
	return multisignatureInfoToIndex;
};
```